ksqldb-serde/src/main/java/io/confluent/ksql/serde/GenericKeySerDe.java

```diff
@@ -148,21 +148,12 @@ private static void checkUnsupportedSchema(
       }
     }
 
-    if (config.getBoolean(KsqlConfig.KSQL_COMPLEX_KEY_FORMAT_ENABLED)) {
-      for (final SimpleColumn column : columns) {
-        if (containsMapType(column.type())) {
-          throw new KsqlException("Map keys, including types that contain maps, are not supported "
-              + "as they may lead to unexpected behavior due to inconsistent serialization. "
-              + "Key column name: " + column.name() + ". "
-              + "Column type: " + column.type().toString(FormatOptions.none()));
-        }
-      }
-    } else {
-      for (final SimpleColumn column : columns) {
-        final SqlType sqlType = column.type();
-        if (!(sqlType instanceof SqlPrimitiveType || sqlType instanceof SqlDecimal)) {
-          throw new KsqlException("Unsupported key schema: " + columns);
-        }
+    for (final SimpleColumn column : columns) {
+      if (containsMapType(column.type())) {
+        throw new KsqlException("Map keys, including types that contain maps, are not supported "
```

Comment:
(from a previous PR) we might want to link to https://github.com/confluentinc/ksql/issues/6621 in the exception of the message if anyone needs to "continue the conversation"

Reply:
Sorry, must've missed the previous comment. Updated in this PR. I went with simply `See https://github.com/confluentinc/ksql/issues/6621 for more.` rather than language about "continuing the conversation" since the latter makes it sound like this is an ongoing discussion which it isn't (at least as of now).

Reply:
ah sorry my comment wasn't clear! I meant (this is code from a previous PR) not (I left a comment on a previous PR). And definitely don't think we should include "continue the conversation" in the exception message! I meant it as that's what they'd be doing.

Reply:
Ah got'cha. Glad we're on the same page :)